changedetectionio/store.py

```diff
@@ -348,20 +348,29 @@ def save_history_text(self, watch_uuid, contents):
         return fname
 
     def sync_to_json(self):
-        print("Saving..")
-        data ={}
+        logging.info("Saving JSON..")
 
         try:
             data = deepcopy(self.__data)
-        except RuntimeError:
-            time.sleep(0.5)
-            print ("! Data changed when writing to JSON, trying again..")
+        except RuntimeError as e:
+            # Try again in 15 seconds
+            time.sleep(15)
+            logging.error ("! Data changed when writing to JSON, trying again.."+str(e))
             self.sync_to_json()
             return
         else:
-            with open(self.json_store_path, 'w') as json_file:
-                json.dump(data, json_file, indent=4)
-                logging.info("Re-saved index")
+
+            try:
+                # Re #286  - First write to a temp file, then confirm it looks OK and rename it
+                # This is a fairly basic strategy to deal with the case that the file is corrupted,
+                # system was out of memory, out of RAM etc
+                with open(self.json_store_path+".tmp", 'w') as json_file:
```

Comment:
I can think of two things here

- Maybe load it in and confirm it works (or check the last two chars are \0A and })
-  How to alert the UI that something 'weird' is going on?
-